payjoin/src/core/receive/v2/session.rs

```diff
@@ -178,36 +178,53 @@ mod tests {
 
     use super::*;
     use crate::persist::test_utils::InMemoryTestPersister;
-    use crate::receive::common::test::unchecked_proposal_from_test_vector;
+    use crate::persist::NoopSessionPersister;
     use crate::receive::tests::original_from_test_vector;
     use crate::receive::v2::test::SHARED_CONTEXT;
     use crate::receive::v2::{
         Initialized, MaybeInputsOwned, PayjoinProposal, ProvisionalProposal, UncheckedProposal,
     };
 
+    fn unchecked_receiver_from_test_vector() -> Receiver<UncheckedProposal> {
+        Receiver {
+            state: UncheckedProposal {
+                original: original_from_test_vector(),
+                session_context: SHARED_CONTEXT.clone(),
+            },
+        }
+    }
+
     #[test]
     fn test_session_event_serialization_roundtrip() {
+        let persister = NoopSessionPersister::<SessionEvent>::default();
+
```

Comment:
The problem here is that  the v1 typestate machine was awkwardly being used to produce the v2 types.

If we made helper transition methods that didn't save and *just* produced errors, we could test without any persister at all, and have the main transition functions call those helpers. But we don't really need that additional boilerplate and can just test using a `NoopSessionPersister`.

I tried to leave expects where there were expects and otherwise just `unwrap()`.